File: php/strategy/ModelDuck.php
<?
namespace strategy;

class ModelDuck extends Duck {
    public function __construct() {
        parent::__construct();
        $this->flyBehavior = new FlyNoWay();
        $this->quackBehavior = new Quack();
    }

    public function display() {
        echo "I'm a model duck";
    }
}
?>

File: java/1.7/strategy/src/com/duck/duck/ModelDuck.java
package com.duck.duck;

import com.duck.flyBehavior.FlyNoWay;
import com.duck.quackBehavior.Quack;

public class ModelDuck extends Duck {
	public ModelDuck() {
		flyBehavior = new FlyNoWay();
		quackBehavior = new Quack();
	}

	public void display() {
		System.out.println("I'm a model duck");
	}
}
